suite_context.go

```diff
@@ -0,0 +1,488 @@
+	}
+	expected := strings.Split(files.Content, "\n")
+	var actual []string
+	for _, ft := range s.testedSuite.features {
+		actual = append(actual, ft.Path)
+	}
+	if len(expected) != len(actual) {
+		return fmt.Errorf("expected %d feature paths to be parsed, but have %d", len(expected), len(actual))
+	}
+	for i := 0; i < len(expected); i++ {
+		var matched bool
+		split := strings.Split(expected[i], "/")
+		exp := filepath.Join(split...)
+		for j := 0; j < len(actual); j++ {
+			split = strings.Split(actual[j], "/")
+			act := filepath.Join(split...)
+			if exp == act {
+				matched = true
+				break
+			}
+		}
+		if !matched {
+			return fmt.Errorf(`expected feature path "%s" at position: %d, was not parsed, actual are %+v`, exp, i, actual)
+		}
+	}
+	return nil
+}
+
+func (s *suiteContext) iRunFeatureSuite() error {
+	if err := s.parseFeatures(); err != nil {
+		return err
+	}
+	s.testedSuite.fmt = testFormatterFunc("godog", &s.out)
+	s.testedSuite.run()
+	s.testedSuite.fmt.Summary()
+
+	return nil
+}
+
+func (s *suiteContext) numScenariosRegistered(expected int) (err error) {
+	var num int
+	for _, ft := range s.testedSuite.features {
+		num += len(ft.ScenarioDefinitions)
+	}
+	if num != expected {
+		err = fmt.Errorf("expected %d scenarios to be registered, but got %d", expected, num)
+	}
+	return
+}
+
+func (s *suiteContext) thereWereNumEventsFired(_ string, expected int, typ string) error {
+	var num int
+	for _, event := range s.events {
+		if event.name == typ {
+			num++
+		}
+	}
+	if num != expected {
+		return fmt.Errorf("expected %d %s events to be fired, but got %d", expected, typ, num)
+	}
+	return nil
+}
+
+func (s *suiteContext) thereWasEventTriggeredBeforeScenario(expected string) error {
+	var found []string
+	for _, event := range s.events {
+		if event.name != "BeforeScenario" {
+			continue
+		}
+
+		var name string
+		switch t := event.args[0].(type) {
+		case *gherkin.Scenario:
+			name = t.Name
+		case *gherkin.ScenarioOutline:
+			name = t.Name
+		}
+		if name == expected {
+			return nil
+		}
+
+		found = append(found, name)
+	}
+
+	if len(found) == 0 {
+		return fmt.Errorf("before scenario event was never triggered or listened")
+	}
+
+	return fmt.Errorf(`expected "%s" scenario, but got these fired %s`, expected, `"`+strings.Join(found, `", "`)+`"`)
+}
+
+func (s *suiteContext) theseEventsHadToBeFiredForNumberOfTimes(tbl *gherkin.DataTable) error {
+	if len(tbl.Rows[0].Cells) != 2 {
+		return fmt.Errorf("expected two columns for event table row, got: %d", len(tbl.Rows[0].Cells))
+	}
+
+	for _, row := range tbl.Rows {
+		num, err := strconv.ParseInt(row.Cells[1].Value, 10, 0)
+		if err != nil {
+			return err
+		}
+		if err := s.thereWereNumEventsFired("", int(num), row.Cells[0].Value); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *suiteContext) theRenderJSONWillBe(docstring *gherkin.DocString) error {
+	loc := regexp.MustCompile(`"suite_test.go:\d+"`)
```

Comment:
when there is -race option, go adds more imports on the fly. and the line differs